Library/Homebrew/utils.rb

```diff
@@ -204,8 +204,8 @@ def puts_columns items, star_items=[]
     # determine the best width to display for different console sizes
     console_width = `/bin/stty size`.chomp.split(" ").last.to_i
     console_width = 80 if console_width <= 0
-    longest = items.sort_by { |item| item.length }.last
-    optimal_col_width = (console_width.to_f / (longest.length + 2).to_f).floor
+    max_len = items.map(&:length).max
```

Comment:
Couldn’t we do it in one pass? Your implementation is still better than the original code but something like this would probably be faster / more memory-efficient:

```ruby
max_len = items.reduce(0) { |max, item| l = item.length ; l > max ? l : max }
```